### DISCUSSION TITLE
How to convert Dialogic.get_variable("something") to an integer or float instead of string?

### DISCUSSION THREAD

**Discussion by creativecorvus3** [dialogic-godot/dialogic#1319] [Q&A] [answered]
I tried many ways to convert this variable into an int or float but it won't work resulting in errors.

str2var()
float()
int()
to_float()
to_int()

These all don't work. So I wanted to know if there is a way to convert this string into another value or is that hard coded?

What I essentially want to do is this
func time():
	if Dialogic.get_variable("time_hh").to_int() >23:
		Dialogic.set_variable("day", +1)
		Dialogic.set_variable("time_hh", 0)
	if int(Dialogic.get_variable("time_mm")) >59:
		Dialogic.set_variable("time_hh", +1)
		Dialogic.set_variable("time_mm", 0)

**Comment by Jowan-Spooner** [comment, marked as answer]
Not sure what the problem is. `if int(Dialogic.get_variable('something')) > 23:` should work.

This depends on your godot version too, I assume you use 3.5. Dialogic.get_variable() should return a normal string.

**Reply by creativecorvus3** [reply]
The error I get is (Invalid Call. Nonexistant 'int' constructor)

Godot Engine ver 3.5.1 stable
Dialogic 1.4.5

**Reply by creativecorvus3** [reply]
I'll just mark this as answered. I fixed it by removing and re-adding dialogic. Weird don't know what made it not work.
